src/as-validator.c

```diff
@@ -1473,6 +1473,15 @@ as_validator_check_relations (AsValidator *validator,
 				as_validator_add_issue (validator, iter, "relation-hardware-value-invalid", content);
 		}
 
+		/* check memory for sanity */
+		if (item_kind == AS_RELATION_ITEM_KIND_MEMORY) {
+			for (guint i = 0; content[i] != '\0'; i++) {
```

Comment:
Instead of this loop you could also just do a `if (g_ascii_strtoll (content, NULL, 10) == 0)`, that would be quite a bit more readable.

Reply:
I have tested this. It let thins like `<memory>8GB</memory>` pass and I haven't find a function that checks if a whole string has only digits.

Reply:
Right, `strtoll` is actually very forgiving. In this case, please use the (brand new!) `as_str_verify_integer (content, 1, G_MAXINT64))` function here :-)

Reply:
I have now this code:
```c
/* check memory for sanity */
if (item_kind == AS_RELATION_ITEM_KIND_MEMORY) {
	if (as_str_verify_integer (content, 1, G_MAXINT64)) {
		as_validator_add_issue (validator, iter, "relation-memory-value-invalid", content);
	}
}
```
But it does not work. It didn't catch anything. I had already merged the latest Upstream commits.

Reply:
This condition needs to be negated - at the moment you are checking if the value is an integer within the set limits and if that is the case, you emit the issue - you do want to do the exact opposite though ;-)

Also, you can remove the extra braces, braces are only added when multiple lines are within the if condition, or if it is followed by an else branch that also contains multiple statements. In any other case, omitting braces if fine and preferred :-)